src/user.controller.ts

```diff
@@ -117,8 +139,108 @@ export class UserController {
+        contentType: 'application/json',
+        statusCode: '200',
+    })
+    public async updateUser(@Body({ type: UpdateUserRequest }) request: UpdateUserRequest): Promise<Response> {
+        const user = await this.repository.createQueryBuilder('user')
+            .where('user.id = :id ', { id: request.id })
+            .getOne();
+
+        if (!user) {
+            return {
+                success: false,
+                message: 'User not found',
+            }
+        }
+        user.firstName = request.firstName;
+        user.lastName = request.lastName;
+        try {
+            const user_updated = await this.repository.save(user);
+            return {
+                success: true,
+                id: user_updated.id,
+                message: 'User Updated',
+            }
+        } catch (error) {
+            return {
+                success: false,
+                message: error,
+            }
+        }
     }
 
+    @Get('/enable/:key')
+    @ContentType("application/json")
+    @OpenAPI({ summary: 'Enable an user' })
+    @ResponseSchema(Response, {
+        contentType: 'application/json',
+        statusCode: '200',
+    })
+    public async enableUser(@Param("key") key: string): Promise<Response> {
+        try {
+            const hashUserRepo = getManager().getRepository(HashUser);
+            const user_hash: any = await hashUserRepo.createQueryBuilder('hashuser')
+                .innerJoinAndSelect("hashuser.user", "user")
+                .where("hashuser.useDate is null and hashuser.creationDate + (:validationTime||' hour')::interval >= current_timestamp(0) and hashuser.key = :key ",
+                    { validationTime: process.env["HASH_EXPIRE_TIME"], key: key })
+                .getOne();
+            if (!user_hash) {
+                return {
+                    success: false,
+                    message: "The user can not be enabled!",
+                }
+            }
+            user_hash.user.enabled = true;
+            user_hash.useDate = new Date();
+            await getManager().transaction(async transactionalEntityManager => {
+                const savedUser: any = await transactionalEntityManager.getRepository(User).save(user_hash.user);
+                await transactionalEntityManager.getRepository(HashUser).save(user_hash);
+            });
+            return {
+                success: true,
+                id: user_hash.user.id,
+                message: 'User Enabled',
+            }
+        } catch (error) {
+            return {
+                success: false,
+                message: error,
+            }
+        }
+    }
 
+    @Get('/disable/:id')
+    @ContentType("application/json")
+    @OpenAPI({ summary: 'Disable an user' })
+    @ResponseSchema(Response, {
+        contentType: 'application/json',
+        statusCode: '200',
+    })
+    public async disableUser(@Param('id') id: number): Promise<Response> {
+        const user = {
```

Comment:
We can use the spread operator here

Reply:
Can't be posible, we don't recieve the request user in the @Param, we only obtain the ID on the URL.